.github/workflows/build.yml

```diff
@@ -34,3 +34,60 @@ jobs:
         uses: codecov/codecov-action@v1
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: 'v12.18.4'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Build lib
+        run: |
+          yarn install
+          yarn build
+
+      - name: Rebuild production node_modules
+        run: |
+          yarn install --production
+          ls node_modules
+
+      - name: artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: |
+            lib
+            node_modules
+
+  integration:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: 'v12.18.4'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: dist
+
+      - name: Deploy
+        uses: ./
+        with:
+          FOLDER: integration
+          BASE_BRANCH: ${{ github.sha }}
+          BRANCH: gh-pages-pr
```

Comment:
We should consider adding a cleanup step here in the off chance that it does actually generate these branches (ie if someone with adequate permissions made a PR). This is a potential window for abuse so it's probably best we prevent things from lingering: https://github.com/JamesIves/github-pages-deploy-action/blob/dev/.github/workflows/integration.yml#L31-L35